More tests for new feature try

thibaultduponchelle opened the pull request:
Hello,

This is some more tests for the try/catch feature 😀 

I tried to stress a bit the feature with some (kind of) real use cases (signal, eval, propagation).

thibaultduponchelle committed:
More tests for new feature try

thibaultduponchelle commented:
@leonerd 😃

jkeenan commented:
> Hello,
> 
> This is some more tests for the try/catch feature grinning
> 
> I tried to stress a bit the feature with some (kind of) real use cases (signal, eval, propagation).

@leonerd, can you review this p.r.?  I'm inclined to apply it within 7 days unless you indicate otherwise.

leonerd commented on t/op/try.t:
```diff
@@ -74,6 +74,88 @@ no warnings 'experimental::try';
     is(f(), "return inside try", 'return inside try');
 }
 
+no warnings "exiting";
```
Move this inside the scope brackets of the test it applies to

thibaultduponchelle replied:
done

leonerd commented on t/op/try.t:
```diff
@@ -74,6 +74,88 @@ no warnings 'experimental::try';
+        }
+    }
+
+    is($result, "d", 'eval last block outside try');
```
ITYM "inside try"

haarg replied:
The block that the last is terminating is outside the try.

thibaultduponchelle replied:
I edited the descriptions to make them clearer

leonerd reviewed (changes requested):
Needs a couple of small changes but otherwise looks good.

jkeenan commented:
@thibaultduponchelle , could you update your pull request based on the feedback ^^ from @leonerd and @haarg ?  Thanks.

thibaultduponchelle commented:
Thank you @leonerd and @haarg for the review. I updated according your remarks 👍🏼

jkeenan commented:
> Thank you @leonerd and @haarg for the review (and thank you @jkeenan). I updated according to the remarks 👍🏼

@leonerd and @haarg, can you review the contributor's most recent changes in this pull request?  Thanks.

leonerd commented:
Overall no huge objection, though in parts I'm unsure quite the value being added by the tests. While "more tests are always good", there gets to a point where we're just testing many minor variations on the same idea, which of course adds size and time.

The signal one at the end, in particular, doesn't actually test signal propagation *through* a `try` boundary, only entirely within on.

jkeenan commented:
> 
> The signal one at the end, in particular, doesn't actually test signal propagation _through_ a `try` boundary, only entirely within on.

@thibaultduponchelle, can you respond to this point raised by @leonerd ?

Thank you very much.
Jim Keenan

thibaultduponchelle commented:
@jkeenan I’m going to do the change 👍🏼

thibaultduponchelle commented:
@jkeenan @leonerd I moved SIGINT handler out of `try {}` block and added an outer and inner block to make it clearer that correct `catch` is reached.
